pkg/statement/statement.go

```diff
@@ -150,6 +152,13 @@ func (a *AbstractStatement) AlgorithmInplaceConsideredSafe() error {
 			ast.AlterTableTruncatePartition,
 			ast.AlterTableAddPartitions:
 			continue
+		case ast.AlterTableModifyColumn, ast.AlterTableChangeColumn:
+			// Only safe if changing length of a VARCHAR column. We don't know the type of the column
+			// or its length, so we cannot determine if this is safe only by parsing. We can simply try
+			// INPLACE, and if it fails we will retry with our own schema change process.
+			if spec.NewColumns[0].Tp != nil && spec.NewColumns[0].Tp.GetType() == mysql.TypeVarchar {
```

Comment:
can `spec.NewColumns` ever have more than one element?

Reply:
No. This is inside a loop over `alterStmt.Specs`, so we are looking separately at each clause in the `ALTER TABLE` statement. A `MODIFY` or `CHANGE` clause can only operate on a single column:

```
  | CHANGE [COLUMN] old_col_name new_col_name column_definition
        [FIRST | AFTER col_name]

  | MODIFY [COLUMN] col_name column_definition
        [FIRST | AFTER col_name]

```